contracts/SnapshotToken/Extensions/SimpleVote.sol

```diff
@@ -0,0 +1,191 @@
+        // solium-enable indentation
+
+        incrementVote(proposalId, voteInFavor, voter);
+    }
+
+    function batchRelayedVotes(
+        uint256[] proposalIndices,
+        bool[] votePreferences,
+        address[] voters,
+        bytes32[] r,
+        bytes32[] s,
+        uint8[] v
+    )
+        public
+    {
+        require(
+            proposalIndices.length == votePreferences.length &&
+            votePreferences.length == voters.length &&
+            voters.length == r.length &&
+            r.length == s.length &&
+            s.length == v.length,
+            "Invalid voting arguments"
+        );
+        for (uint i = 0; i < voters.length; i++) {
+            relayedVote(
+                proposalIndices[i],
+                votePreferences[i],
+                voters[i],
+                r[i],
+                s[i],
+                v[i]
+            );
+        }
+    }
+
+    /// @return whether the vote on a proposal has passed, if false this could be because it was rejected or
+    ///   because the quorum was not reached.
+    /// @dev throws if vote is still ongoing or proposal does not exist. Also deletes the proposalData to free up gas.
+    //     Therefore subsequent attempts to call this function will throw too
+    function getResult(uint256 proposalId) public {
+        require(proposalId < _nextProposalId, "Invalid proposal ID");
+        Proposal storage p = _proposals[proposalId];
+        require(p.owner != address(0), "Proposal was already deleted");
+        require(p.endTime < now, "Voting period is not over");
+
+        uint256 minParticipation = TOKEN.totalSupplyAt(p.snapshotId) * 40 / 100;
+        bool passedQuorum = p.inFavor + p.against > minParticipation;
+        bool hasPassed = passedQuorum && p.inFavor > p.against;
+        _hasActiveProposal[_proposals[proposalId].owner] = false;
+
+        emit LogProposalResult(proposalId, hasPassed, p.inFavor, p.against);
+
+        delete _proposals[proposalId];
+    }
+
+    /// @dev increase the votecount on a given proposal by the token balance of a given address,
+    ///   throws if proposal does not exist or the vote on it has ended already. Votes are final,
+    ///   changing the vote is not allowed
+    /// @param proposalId of the proposal to be voted on
+    /// @param voteInFavor of the desired proposal
+    /// @param voter address whose tokenBalance is to be used as voting-weight
+    function incrementVote(uint256 proposalId, bool voteInFavor, address voter) internal {
+        require(proposalId < _nextProposalId, "Invalid proposal ID");
+        Proposal storage p = _proposals[proposalId];
+        require(p.endTime > now, "Voting period is over");
+        require(!p.hasVoted[voter], "Address has already voted");
+        uint256 weight = TOKEN.balanceOfAt(voter, p.snapshotId);
+        require(weight > 0, "Token balance at proposal time is zero");
+
+        if (voteInFavor) {
+            p.inFavor += weight;
```

Comment:
just one small note (I discovered this recently). you read and write storage here. in that case solidity optimizer will be unable to reduce number of writes to storage. the patterns is to read storage and then write it in one block.

here it does not matter - we use full words to store vote weights